File: 71. Simplify Path.java
import java.util.Stack;

class Solution {
    public String simplifyPath(String path) {
        String dirs[] = path.split("/");
        Stack<String> stack = new Stack<>();

        for(String dir : dirs){
            if(dir.isEmpty() || dir.equals(".")) continue;
            if(dir.equals("..")){
                if(!stack.isEmpty())
                    stack.pop();
            } else{
                stack.add(dir);
            }
        }

        return "/" + String.join("/", stack);
    }
}

File: 71. Simplify Path.py


class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        for dir in path.split("/"):
            if dir in ('', '.'):
                continue
            if dir == '..':
                if stack:
                    stack.pop()
            else:
                stack.append(dir)
        
        return "/" + "/".join(stack)
            